connector/protobuf/src/main/scala/org/apache/spark/sql/protobuf/ProtobufDeserializer.scala

```diff
@@ -299,6 +298,43 @@ private[sql] class ProtobufDeserializer(
     }
   }
 
+  private def getFieldValue(record: DynamicMessage, field: FieldDescriptor): AnyRef = {
+    // We return a value if one of:
+    // - the field is repeated
+    // - the field is explicitly present in the serialized proto
+    // - the field is proto2 with a default
+    // - we should materialize the zero value
+    //
+    // Repeated fields have to be treated separately as they cannot have `hasField`
+    // called on them.
+    if (
+      field.isRepeated
+        || record.hasField(field) // checks for the field explicitly being set
+        || field.hasDefaultValue
+        || shouldGetZeroValue(record, field)) {
+      record.getField(field)
+    } else {
+      null
+    }
+  }
+
+  // If the materializeZeroValues config is set, we will materialize zero values when the field
```

Comment:
Ideally we shouldn't need very long comment. Brief line and pointing to documentation of the flag should be enough. with that, we could move the check back into `getFieldValue()` method.

Reply:
i only added the comment because you expressed some concerns about the original implementation :(. i'll remove it

Reply:
done